mesonbuild/compilers/compilers.py

```diff
@@ -276,6 +276,25 @@ def is_library(fname):
 msvc_debug_args = {False: [],
                    True: []} # Fixme!
 
+arm_optimization_args = {'0': ['-O0'],
+                         'g': ['-g'],
+                         '1': ['-O1'],
+                         '2': ['-O2'],
+                         '3': ['-O3'],
+                         't': ['-Otime'],
+                         }
+
+armclang_optimization_args = {'0': ['-O0'],
+                              'g': ['-g'],
+                              '1': ['-O1'],
+                              '2': ['-O2'],
+                              '3': ['-O3'],
+                              'f': ['-Ofast'],
+                              's': ['-Os'],
+                              'z': ['-Oz'],
+                              't': ['-flto'],
```

Comment:
Why are you adding new optimization types here? Please stick to the existing ones. All optimization types must map to other compilers too, and should be done separately from this commit.

Reply:
The new optimization types are supported in ARM and ARMClang compilers and should be included as they might be needed. 
Are you suggesting that the other compilers should also have these options and must be kept empty to have consistency across compilers?

Reply:
All compilers should have the same options. If you have extras they can not be selected because the optimization option will not accept any other values. Please remove the extra ones so we can merge this for the next release which will happen Saturday. We can look into adding new things later.